src/background/services/zkIdentity/services/ZkIdentityFactory.ts

```diff
@@ -0,0 +1,44 @@
+import { Identity } from "@semaphore-protocol/identity";
+
+import { ICreateIdentityArgs, StrategiesMap } from "@src/types";
+import checkParameter from "@src/util/checkParameter";
+
+import { ZkIdentitySemaphore } from "../protocols/ZkIdentitySemaphore";
+
+export class ZkIdentityFactoryService {
+  protected createNewIdentity = (strategy: keyof StrategiesMap, config: ICreateIdentityArgs): ZkIdentitySemaphore =>
+    this.strategiesMap[strategy](config);
+
+  private createInterrepIdentity = (config: ICreateIdentityArgs): ZkIdentitySemaphore => {
+    const { identityStrategy, web2Provider, name, messageSignature, account } = config;
+
+    checkParameter(messageSignature, "messageSignature", "string");
```

Comment:
I don't think we need this param checks because typescript will fail on compilation in something is wrong